cloudimg/ms_azure.py

```diff
@@ -558,3 +572,32 @@ def tag_image(self, blob_props, tags):
             raise BlobNotFoundError(err)
 
         return blob.set_blob_tags(tags)
+
+    def delete(self, metadata):
+        """
+        Delete VHD images for given metadata.
+
+        Args:
+            metadata(AzureDeleteMetadata): Metadata about the image
+
+        Returns:
+            tuple (image_name[Str], image_path[Str]) of removed objects.
+        """
+        image_name = metadata.image_name
+
+        log.info('Searching for image: %s', image_name)
+        blob = self.get_object_by_name(
+            container=metadata.container,
+            name=image_name
+        )
+
+        if not blob:
+            log.info('Image does not exist: %s', image_name)
+            return None, None
+
+        image_path = "%s/%s" % (blob.container_name, blob.blob_name)
+        log.info("Deleting the image from %s", image_path)
+        blob.delete_blob(delete_snapshots="include")
```

Comment:
It would be nice to add link to documentation.
* https://learn.microsoft.com/en-us/python/api/azure-storage-blob/azure.storage.blob.blobclient?view=azure-python#azure-storage-blob-blobclient-delete-blob

just in case sb would like to find out (from github) what the "include" means :-)

But that is just a nitpick.

Otherwise LGTM

Reply:
Thanks!

I've added it